mining/worker_test.go

```diff
@@ -280,28 +279,20 @@ func TestGenerateMultiBlockTipSet(t *testing.T) {
 		MessageStore:  messages,
 	}, CreatePoSTFunc)
 
-	parents := types.NewTipSetKey(newCid())
-	stateRoot := newCid()
-	baseBlock1 := types.Block{
-		Parents:      parents,
-		Height:       types.Uint64(100),
-		ParentWeight: types.Uint64(1000),
-		StateRoot:    stateRoot,
-	}
-	baseBlock2 := types.Block{
-		Parents:      parents,
-		Height:       types.Uint64(100),
-		ParentWeight: types.Uint64(1000),
-		StateRoot:    stateRoot,
-		Nonce:        1,
-	}
-	blk, err := worker.Generate(ctx, th.RequireNewTipSet(t, &baseBlock1, &baseBlock2), nil, types.PoStProof{}, 0)
+	builder := chain.NewBuilder(t, address.Undef)
+	genesis := builder.NewGenesis()
+
+	parentTipset := builder.AppendManyOn(99, genesis)
+	baseTipset := builder.AppendOn(parentTipset, 2)
+	assert.Equal(t, 2, baseTipset.Len())
+
+	blk, err := worker.Generate(ctx, baseTipset, nil, types.PoStProof{}, 0)
 	assert.NoError(t, err)
 
 	assert.Equal(t, types.EmptyMessagesCID, blk.Messages)
 	assert.Equal(t, types.EmptyReceiptsCID, blk.MessageReceipts)
 	assert.Equal(t, types.Uint64(101), blk.Height)
-	assert.Equal(t, types.Uint64(1020), blk.ParentWeight)
+	assert.Equal(t, types.Uint64(120), blk.ParentWeight)
```

Comment:
I would slightly prefer if you set the baseTipset's weight after constructing with the builder, this way the internals of the builder's testing weight function are not needed to understand the test.

Reply:
If that involved mutating the blocks after construction, it's probably a step backwards. I think this is ok for now. A configurable weight function in the builder would be better, but probably not worth it here.